The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/14025263416/job/39262803741) [(plain)](https://github.com/rust-lang/rust/commit/4615eed0f477c95c38a5a0f7af4f7153ef67ade0/checks/39262803741/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#19 exporting to docker image format
#19 sending tarball 21.7s done
#19 DONE 34.9s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
[CI_JOB_NAME=x86_64-gnu-llvm-18]
debug: `DISABLE_CI_RUSTC_IF_INCOMPATIBLE` configured.
---
sccache: Listening on address 127.0.0.1:4226
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'build.print-step-timings', '--enable-verbose-tests', '--set', 'build.metrics', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: build.build          := x86_64-unknown-linux-gnu
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
[RUSTC-TIMING] object test:false 2.702
error[E0425]: cannot find value `opened_devnull` in this scope
   --> library/std/src/sys/pal/unix/mod.rs:70:20
    |
63  | /         macro_rules! open_devnull {
64  | |             ($opened_devnull:expr) => {
65  | |                 #[cfg(not(all(target_os = "linux", target_env = "gnu")))]
66  | |                 use libc::open as open64;
...   |
70  | |                 if opened_devnull != -1 {
    | |                    ^^^^^^^^^^^^^^ not found in this scope
...   |
84  | |         }
    | |_________- in this expansion of `open_devnull!`
...
129 |                   open_devnull!(opened_devnull);
    |                   ----------------------------- in this macro invocation

error[E0425]: cannot find value `opened_devnull` in this scope
   --> library/std/src/sys/pal/unix/mod.rs:71:34
    |
63  | /         macro_rules! open_devnull {
64  | |             ($opened_devnull:expr) => {
65  | |                 #[cfg(not(all(target_os = "linux", target_env = "gnu")))]
66  | |                 use libc::open as open64;
...   |
71  | |                     if libc::dup(opened_devnull) != -1 {
    | |                                  ^^^^^^^^^^^^^^ not found in this scope
...   |
84  | |         }
    | |_________- in this expansion of `open_devnull!`
...
129 |                   open_devnull!(opened_devnull);
    |                   ----------------------------- in this macro invocation

error[E0425]: cannot find value `opened_devnull` in this scope
   --> library/std/src/sys/pal/unix/mod.rs:75:17
    |
63  | /         macro_rules! open_devnull {
64  | |             ($opened_devnull:expr) => {
65  | |                 #[cfg(not(all(target_os = "linux", target_env = "gnu")))]
66  | |                 use libc::open as open64;
...   |
75  | |                 opened_devnull = open64(c"/dev/null".as_ptr(), libc::O_RDWR, 0);
    | |                 ^^^^^^^^^^^^^^ not found in this scope
...   |
84  | |         }
    | |_________- in this expansion of `open_devnull!`
...
129 |                   open_devnull!(opened_devnull);
    |                   ----------------------------- in this macro invocation

error[E0425]: cannot find value `opened_devnull` in this scope
   --> library/std/src/sys/pal/unix/mod.rs:76:20
    |
63  | /         macro_rules! open_devnull {
64  | |             ($opened_devnull:expr) => {
65  | |                 #[cfg(not(all(target_os = "linux", target_env = "gnu")))]
66  | |                 use libc::open as open64;
...   |
76  | |                 if opened_devnull == -1 {
    | |                    ^^^^^^^^^^^^^^ not found in this scope
...   |
84  | |         }
    | |_________- in this expansion of `open_devnull!`
...
129 |                   open_devnull!(opened_devnull);
    |                   ----------------------------- in this macro invocation

error[E0425]: cannot find value `opened_devnull` in this scope
   --> library/std/src/sys/pal/unix/mod.rs:70:20
    |
63  | /         macro_rules! open_devnull {
64  | |             ($opened_devnull:expr) => {
65  | |                 #[cfg(not(all(target_os = "linux", target_env = "gnu")))]
66  | |                 use libc::open as open64;
...   |
70  | |                 if opened_devnull != -1 {
    | |                    ^^^^^^^^^^^^^^ not found in this scope
...   |
84  | |         }
    | |_________- in this expansion of `open_devnull!`
...
150 |                       open_devnull!(opened_devnull);
    |                       ----------------------------- in this macro invocation

error[E0425]: cannot find value `opened_devnull` in this scope
   --> library/std/src/sys/pal/unix/mod.rs:71:34
    |
63  | /         macro_rules! open_devnull {
64  | |             ($opened_devnull:expr) => {
65  | |                 #[cfg(not(all(target_os = "linux", target_env = "gnu")))]
66  | |                 use libc::open as open64;
...   |
71  | |                     if libc::dup(opened_devnull) != -1 {
    | |                                  ^^^^^^^^^^^^^^ not found in this scope
...   |
84  | |         }
    | |_________- in this expansion of `open_devnull!`
...
150 |                       open_devnull!(opened_devnull);
    |                       ----------------------------- in this macro invocation

error[E0425]: cannot find value `opened_devnull` in this scope
   --> library/std/src/sys/pal/unix/mod.rs:75:17
    |
63  | /         macro_rules! open_devnull {
64  | |             ($opened_devnull:expr) => {
65  | |                 #[cfg(not(all(target_os = "linux", target_env = "gnu")))]
66  | |                 use libc::open as open64;
...   |
75  | |                 opened_devnull = open64(c"/dev/null".as_ptr(), libc::O_RDWR, 0);
    | |                 ^^^^^^^^^^^^^^ not found in this scope
...   |
84  | |         }
    | |_________- in this expansion of `open_devnull!`
...
150 |                       open_devnull!(opened_devnull);
    |                       ----------------------------- in this macro invocation

error[E0425]: cannot find value `opened_devnull` in this scope
   --> library/std/src/sys/pal/unix/mod.rs:76:20
    |
63  | /         macro_rules! open_devnull {
64  | |             ($opened_devnull:expr) => {
65  | |                 #[cfg(not(all(target_os = "linux", target_env = "gnu")))]
66  | |                 use libc::open as open64;
...   |
76  | |                 if opened_devnull == -1 {
    | |                    ^^^^^^^^^^^^^^ not found in this scope
...   |
84  | |         }
    | |_________- in this expansion of `open_devnull!`
...
150 |                       open_devnull!(opened_devnull);
    |                       ----------------------------- in this macro invocation

For more information about this error, try `rustc --explain E0425`.
[RUSTC-TIMING] std test:false 1.796
error: could not compile `std` (lib) due to 8 previous errors
```

</details>

--- a/library/std/src/sys/pal/unix/mod.rs
+++ b/library/std/src/sys/pal/unix/mod.rs
@@ -61,19 +61,21 @@ pub unsafe fn init(argc: isize, argv: *const *const u8, sigpipe: u8) {
     }
 
     unsafe fn sanitize_standard_fds() {
+        #[allow(dead_code)]
         let mut opened_devnull = -1;
+        #[allow(dead_code)]
         let mut open_devnull = || {
             #[cfg(not(all(target_os = "linux", target_env = "gnu")))]
-            use libc::open as open64;
+            use libc::open;
             #[cfg(all(target_os = "linux", target_env = "gnu"))]
-            use libc::open64;
+            use libc::open64 as open;
 
             if opened_devnull != -1 {
                 if libc::dup(opened_devnull) != -1 {
                     return;
                 }
             }
-            opened_devnull = open64(c"/dev/null".as_ptr(), libc::O_RDWR, 0);
+            opened_devnull = open(c"/dev/null".as_ptr(), libc::O_RDWR, 0);
             if opened_devnull == -1 {
                 // If the stream is closed but we failed to reopen it, abort the
                 // process. Otherwise we wouldn't preserve the safety of

